code/drasil-lang/lib/Language/Drasil/Expr/Lang.hs

```diff
@@ -282,6 +301,12 @@ instance Typed Expr Space where
               (\_ -> all (\ r -> length r == columns && all (== expT) r) sss)
               (const False) expT
         t = fromLeft (error "Infer on Matrix had a strong expectation of Left-valued data.") expT -- This error should never occur.
+        
+  infer cxt (Set e) =
+    case infer cxt e of
+        Left sp -> if S.isBasicNumSpace sp then Left sp else Right (show sp)
```

Comment:
I'm not sure if we need to restrict ourselves to only numeric sets. For example, power sets and string sets are probably things we want. However, we can probably just admit anything in the set, so as long as the elements are of the same type.

Reply:
Yes, this was left in by accident.